iroh-sync/src/ranger.rs

```diff
@@ -568,6 +570,30 @@ where
                     start_index += 1;
                 }
 
+                // TODO: I had troubles understanding the split algorithm above and rewrote it to
```

Comment:
please look at the new version from @rklaehn, which is very likely more correct than the implemented one